app_dart/lib/src/request_handlers/update_task_status.dart

```diff
@@ -58,19 +58,22 @@ class UpdateTaskStatus extends ApiRequestHandler<UpdateTaskStatusResponse> {
 
     final Commit commit = await config.db.lookupValue<Commit>(task.commitKey);
 
+    final int endTimestamp = DateTime.now().millisecondsSinceEpoch;
     if (newStatus == Task.statusFailed) {
       // Attempt to de-flake the test.
       final int maxRetries = await config.maxTaskRetries;
       if (task.attempts >= maxRetries) {
         task.status = Task.statusFailed;
         task.reason = 'Task failed on agent';
+        task.endTimestamp = endTimestamp;
       } else {
         // This will cause this task to be picked up by an agent again.
         task.status = Task.statusNew;
         task.startTimestamp = 0;
       }
     } else {
       task.status = newStatus;
+      task.endTimestamp = endTimestamp;
```

Comment:
I'd prefer making the call inline: `task.endTimestamp = DateTime.now().millisecondsSinceEpoch;`.
Also applies to line 68, and remove the `endTimestamp` variable.

Reply:
Updated.